### ISSUE TITLE
Migrate MediaService to androidx.media2 for paging in Android Auto

### ISSUE THREAD

**Issue by Maxr1998** [jellyfin/jellyfin-android#615]
Using [`MediaLibraryService`](https://developer.android.com/reference/androidx/media2/session/MediaLibraryService) instead of `MediaBrowserCompat`, we can return pages of data, which should fix #284.

**Comment by kaysond** [comment]
@Maxr1998 any current thoughts on using media2 or media3? Looks like there's already an alpha version of 3, so it'll probably release by end of the year. Might be an option to work on media3 in a separate branch as a beta release track?

I'd be interested in working on this, though to be honest, it's been a long time since I did anything android-related.

**Comment by Maxr1998** [comment]
I first wanted to upgrade ExoPlayer to the latest version and finally finish and merge the transcoding selection in #535.
Afterwards, we could indeed try migrating to media 3.

**Comment by foss-** [comment]
Seeing that ExoPlayer was updated to 2.19.1 in 2.6.0 and #535 being merged a while ago, are there plans for the media3 migration? Couldn't find a separate issue so asking here.

**Comment by nielsvanvelzen** [comment]
I made the media3 migration in #1118 but @Maxr1998 wanted to do some other stuff first so it was postponed.

**Comment by Maxr1998** [comment]
Yes, I want to get in the UI rewrite first but I haven't been able to work on it recently. Too busy with work and university, unfortunately.

**Comment by channeladam** [comment]
FYI - the latest media3 (release v1.2.1 on 17 January 2024 - https://github.com/androidx/media/releases/tag/1.2.1) finally fixes the annoying 3 year old bug with the OPUS and VORBIS channel layouts being wrong for 3, 5, 6, 7 and 8 channels (https://github.com/google/ExoPlayer/issues/8396).

It would be awesome to have that!

**Comment by channeladam** [comment]
@nielsvanvelzen I happened to notice the Android TV Roadmap (https://github.com/orgs/jellyfin/projects/47/views/1) and saw that exoplayer has been replaced with media3 there
(https://github.com/jellyfin/jellyfin-androidtv/pull/2826)

I don't know much about how the Jellyfin code changes are coordinated - but I was wondering if there was an update regarding the Jellyfin Android app and media3 ?  (since it apparently fixes the OPUS and VORBIS channel layout issue)

Thank you!

**Comment by nielsvanvelzen** [comment]
Eventually the mobile app will share the playback code with the Android TV app (the plan is to unify both apps). Until that happens the mobile and TV app will have separate playback implementations with different feature sets.
I worked on the media3 migration a while back (#1118) but it was put on hold (as mentioned above). No current ETA on when we'll upgrade.

**Comment by joselynerodrigueztorres2-ai** [comment]
#1118
